File: frontend/src/lib/types/ledger.ts
export type LedgerRole = "owner" | "admin" | "editor" | "viewer";

export interface Ledger {
  id: number;
  name: string;
  description: string;
  currency: string;
  ownerId: number;
  isDefault: boolean;
  status: number;
  createdAt: string;
  updatedAt: string;
}

export interface LedgerMember {
  id: number;
  ledgerId: number;
  userId: number;
  role: LedgerRole;
  nickname?: string;
  avatar?: string;
  joinedAt: string;
}

export interface CreateLedgerDTO {
  name: string;
  description?: string;
  currency?: string;
}

File: backend/src/main/java/com/mamoji/finance/domain/Ledger.java
package com.mamoji.finance.domain;

/** Finance-owned ledger aggregate state. */
public class Ledger {
    public long id;
    public long companyId;
    public String name;
    public String description;
    public String currency;
    public long ownerId;
    public boolean isDefault;
    public int status;
    public String createdAt;
    public String updatedAt;
}
